Assets/Scripts/GameState/LeaderboardData.cs

```diff
@@ -70,47 +68,49 @@ public static void SaveScores(List<Highscore> scores)
     /// <returns>An ordered list of <c>Highscore</c> structs.</returns>
     public static List<Highscore> LoadScores()
     {
-        if (File.Exists(path))
+        if (!File.Exists(highscoresDataPath))
+            return CreateMockLeaderboard();
+
+        BinaryFormatter formatter = new BinaryFormatter();
+        FileStream stream = new FileStream(highscoresDataPath, FileMode.Open);
+
+        List<Highscore> data;
+
+        // Check if the file is corrupted
+        try
+        {
+            data = (List<Highscore>)formatter.Deserialize(stream);
+        } catch (SystemException e) when (e is SerializationException || e is InvalidCastException)
```

Comment:
I tested with a corrupted highscores.data file and got `IOException shar violation on path C:/mypath...etc.../highscores.data System.IO.File.Delete (System.String path)`

The code following ` File.Delete(highscoresDataPath);` in the catch block:
           `Debug.LogWarning($"Highscore data was corrupted, it has been replaced.\nException message: {e.Message}");
            return LoadScores();`
Did not run afterwards.

Reply:
Hm, did it say which line it originated from? 😮 Based on that the exception message contains `System.IO.File.Delete`, it looks like the exception came from within the `catch` block, specifically this line: `File.Delete(highscoresDataPath);` 🤔

Reply:
Well, if deleting the file within the catch block fails, then it's only natural that the call to `LogWarning()` does not run 😅 But yeah, after a google search, it looks like the error is caused by the stream not being closed when deleting the file 🤔 I'll fix it in a commit, so thanks for reporting this! 😄

Reply:
Alright, would you mind testing that the file is properly deleted after the latest commit I pushed? 😊

Reply:
Can confirm it works! :)